samples/hello/README.rst

```diff
@@ -184,6 +228,8 @@ This is the output from the serial console:
 Responses to Hello messages are printed above as a hexdump of "Hello mark". This
 means that communication with Golioth is working.
 
+.. _authentication methods: https://docs.golioth.io/firmware/zephyr-device-sdk/authentication/
```

Comment:
No strong opinion here... but I'm a bit afraid of circular referencing (docs are written based this SDK, while here we start to reference those docs), since this page is strictly Golioth Zephyr SDK (this repo) related. It might quickly get outdated.

Reply:
Good call out.